Tools/DataStorage/DataStorage.cs

```diff
@@ -10,6 +10,355 @@ namespace BD_oneLove.Tools.DataStorage
+            SqlConnection myConn = new SqlConnection(StationManager.ConnectionString);
+            try
+            {
+                myConn.Open();
+                int res = 0;
+                using (SqlCommand command = new SqlCommand("set ANSI_WARNINGS  OFF;", myConn))
+                {
+                    command.ExecuteNonQuery();
+                }
+
+
+                using (SqlCommand command = new SqlCommand(sql, myConn))
+                {
+                    res = command.ExecuteNonQuery();
+                }
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("There's problem with you connection!\n" + ex.Message);
+                return false;
+            }
+            finally
+            {
+                myConn?.Close();
+            }
+            return true;
+        }
+
+        public bool UpdateClass(Class c)
+        {
+            string sql = "UPDATE classes " +
+                  $"SET number='{c.Number}', " +
+                  $" letter = '{c.Letter}', " +
+                  $" st_year = '{c.StYear}' "  +
+                  $"WHERE class_id ='{c.ClassId}';";
+            SqlConnection myConn = new SqlConnection(StationManager.ConnectionString);
+            try
+            {
+                myConn.Open();
+                int res = 0;
+                using (SqlCommand command = new SqlCommand("set ANSI_WARNINGS  OFF;", myConn))
+                {
+                    command.ExecuteNonQuery();
+                }
+
+                using (SqlCommand command = new SqlCommand(sql, myConn))
+                {
+                    res = command.ExecuteNonQuery();
+                }
+            }
+            catch (Exception ex)
+            {
+                 MessageBox.Show("There's problem with you connection!\n" + ex.Message);
+                return false;
+            }
+            finally
+            {
+                myConn?.Close();
+
+            }
+            return true;
+
+
+        }
+
+
+
+        public List<Teacher> GetTeachers(Class c)
+        {
+            string sql =
+               "SELECT ht.tab_number, ht.h_name, ht.patronymic, ht.surname" +
+               " FROM (head_teachers ht INNER JOIN head_teachers_classes htc ON ht.tab_number=htc.tab_number) " +
+               $"WHERE htc.class_id='{c.ClassId}'";
+
+            List<Teacher> res = new List<Teacher>();
+            SqlConnection myConn = new SqlConnection(StationManager.ConnectionString);
+            try
+            {
+                myConn.Open();
+
+                using (SqlCommand command = new SqlCommand(sql, myConn))
+                {
+                    var reader = command.ExecuteReader();
+                    while (reader.Read())
+                    {
+                        Teacher cur = new Teacher(reader.GetString(0));
+                        cur.HName = reader.GetString(1);
+                        cur.Patronymiс = reader.GetString(2);
```

Comment:
You know, this might be a trouble. If teacher has no patronymic, which we allow in DB, you will get an Exception here (provided you recorded NULL, not empty string ""). I think we will need to solve this issue, after DB redesign.